### PR TITLE
fix: community buttons

### LINKED ISSUE #17559: Design review: Communities/Join/Action-Buttons
# Bug Report

[Figma File](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=6117-85237&mode=design&t=LkXZMJM46LxxBKYM-0)

**Related issue** #17560

## Issues:

**Action buttons**

- [ ] The action button is not fixed to the bottom as intended
[View Figma Design](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=5963:45900&mode=design#567165943)

- [ ] The color of the "Request to Join" button does not match the design specification
[View Figma Design)](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=5963:45900&mode=design#567166224)

- [ ] Margin bottom/spacing do not align with the design
[View Figma Design](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=5963:45900&mode=design#567175045)
 
### Reproduction

Navigate to Communities-> Join a community

### PR DESCRIPTION
**Pull request by BalogunofAfrica**
Fixes https://github.com/orgs/status-im/projects/90/views/1?groupedBy%5BcolumnId%5D=&filterQuery=is:issue+is:open&pane=issue&itemId=45784928

- Android
- iOS

#### Areas that maybe impacted
- community request 

##### Functional

- community

### Before and after screenshots comparison
Before:
simulator_screenshot_776CD0F6-9189-4476-BBDF-6E7BA72AC31A

After:
simulator_screenshot_0F8D76C7-0E0C-4E8A-A028-1F0032D78E0F

Figma link:
https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=6117-85237&mode=design&t=VrDYIFW0pOuVnDbb-0
simulator_screenshot_28B21662-2AF0-4044-BF26-B0CBABA608E4

status: ready

### CODE REVIEW

File: src/status_im/contexts/communities/actions/community_rules_list/view.cljs [hunk 1]
```diff
@@ -44,9 +44,10 @@
     (str title ": " content)]])
 
 (defn view
-  [rules]
+  [rules scroll-enabled]
```
Comment (J-Son89) [inline src/status_im/contexts/communities/actions/community_rules_list/view.cljs]: consider appending "?" -> `scroll-enabled?`

File: src/status_im/contexts/communities/actions/request_to_join/view.cljs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 (ns status-im.contexts.communities.actions.request-to-join.view
   (:require
     [quo.core :as quo]
+    [quo.foundations.colors :as colors]
+    [quo.theme :as quo.theme]
```
Comment (J-Son89) [inline src/status_im/contexts/communities/actions/request_to_join/view.cljs]: can be ` [quo.theme]`

**Review by jo-mut — APPROVED** [review]

File: src/status_im/contexts/communities/actions/community_rules_list/style.cljs [hunk 3]
```diff
@@ -7,8 +7,8 @@
    :margin-top     16})
 
 (def community-rule-index
-  {:margin-left 5})
+  {:margin-left 4})
 
 (def community-rule-text
-  {:margin-left 6
+  {:margin-left 0
```
Comment (J-Son89) [inline src/status_im/contexts/communities/actions/community_rules_list/style.cljs]: can we just remove this?

File: src/status_im/contexts/communities/actions/request_to_join/view.cljs [hunk 4]
```diff
@@ -33,7 +36,7 @@
             :weight              :semi-bold
             :size                :heading-2}
            (i18n/label :t/request-to-join)]]
-         [rn/view {:style {:margin-right :auto :margin-top 8}}
+         [rn/view {:style {:margin-right :auto :margin-top 4}}
```
Comment (J-Son89) [inline src/status_im/contexts/communities/actions/request_to_join/view.cljs]: maybe better to put in a style file?

File: src/status_im/contexts/communities/actions/request_to_join/view.cljs [hunk 5]
```diff
@@ -45,21 +48,25 @@
...
+          :on-press            #(join-community-and-navigate-back id)
+          :container-style     {:flex 1}
+          :inner-style         {:background-color (colors/resolve-color color theme)}}
+         (i18n/label :t/request-to-join)]]
+       [rn/view {:style style/final-disclaimer-container}
+        [quo/text
+         {:size  :paragraph-2
+          :style style/final-disclaimer-text}
+         (i18n/label :t/request-to-join-disclaimer)]]])))
+
```
Comment (J-Son89) [inline src/status_im/contexts/communities/actions/request_to_join/view.cljs:70]: extra line here

File: src/status_im/navigation/screens.cljs [hunk 6]
```diff
@@ -97,7 +97,7 @@
 
     {:name      :community-requests-to-join
      :options   {:sheet? true}
-     :component join-menu/request-to-join}
+     :component join-menu/view}
```
Comment (J-Son89) [inline src/status_im/navigation/screens.cljs:100]: 🙌

**Review by J-Son89 — APPROVED** [review]

**Review by Francesca-G — APPROVED** [review]
It looks good compared to the feedback given in the 1.25 build.

I think the screen was updated in the past few months, please have a look here :)
https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15331%3A285065&mode=design&t=BQtnGwV6YnF1RMeW-1

**Comment by BalogunofAfrica** [comment]
> It looks good compared to the feedback given in the 1.25 build.
> 
> I think the screen was updated in the past few months, please have a look here :) https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15331%3A285065&mode=design&t=BQtnGwV6YnF1RMeW-1


You are right, I confirmed that there are new screens for the request to join and it's currently in the works.

I'd have to close this PR.

cc: @ajayesivan

**Review by ilmotta — APPROVED** [review]
LGTM @BalogunofAfrica 

Indeed, we are working on new screens behind a feature flag for requesting to join communities. But given the PR is already approved by @Francesca-G and the code looks good, I think it's fine to merge and not close this PR. We don't know exactly when we will enable the toggle `community-accounts-selection-enabled?` in `develop`, it might still take a few weeks.

**Comment by BalogunofAfrica** [comment]
> LGTM @BalogunofAfrica
> 
> Indeed, we are working on new screens behind a feature flag for requesting to join communities. But given the PR is already approved by @Francesca-G and the code looks good, I think it's fine to merge and not close this PR. We don't know exactly when we will enable the toggle `community-accounts-selection-enabled?` in `develop`, it might still take a few weeks.

That makes sense

**Pull request merged**